### PR TITLE
Jared Hobson

### PR DESCRIPTION
**Pull request by JaredMHobson**
I found this pretty easy. It was cool finding out about the `maximum` in capybara. Though I didn't do this, I felt like I could have just copy and pasted the entire practice mid mod into this. 

Finished all user stories and the extension.

### CODE REVIEW

File: app/controllers/actor_movies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ActorMoviesController < ApplicationController
+  def create
+    ActorMovie.create(movie_id: params[:movie_id], actor_id: params[:actor_id])
```
Comment (cjsim89) [inline app/controllers/actor_movies_controller.rb:3]: I recommend always using the `!` bang operator to make sure any errors that occur with `.create!` actually bubble up and we're able to use the related validation messages that come along with it. Think of `!` as "failing loudly".

Comment (cjsim89) [reply app/controllers/actor_movies_controller.rb:3]: That said, I like that you created this controller for this user story and used the appropriate action!

File: app/models/movie.rb [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
   belongs_to :studio
+  has_many :actor_movies
+  has_many :actors, through: :actor_movies
+
+  def youngest_to_oldest_actors
+    actors.order(:age)
+  end
+
+  def average_actor_age
+    actors.average(:age).round(2)
```
Comment (cjsim89) [inline app/models/movie.rb:11]: Good AR on both of these methods!

File: app/views/movies/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+<div id="movie_actors">
+  <h2>Actors</h2>
+  <% @movie.youngest_to_oldest_actors.each do |actor| %>
+    <p><%= actor.name %></p>
+  <% end %>
+
+  <h3>Average Actor Age: <%= @movie.average_actor_age %></h3>
+
+  <div id="add_movie_actor">
+    <%= form_with url: "/movies/#{@movie.id}/actors", method: :post, data: { turbo: false }, local: true do |f| %>
```
Comment (cjsim89) [inline app/views/movies/show.html.erb:14]: Nice path & method setup, and form overall!

File: app/models/studio.rb [hunk 4]
```diff
@@ -1,3 +1,8 @@
 class Studio < ApplicationRecord
   has_many :movies
+  has_many :actors, through: :movies
+
+  def unique_actors_list
+    actors.distinct
```
Comment (cjsim89) [inline app/models/studio.rb:6]: Nice use of `.distinct`, as well as that handy association! 🙌

File: app/views/movies/show.html.erb (hunk 3 above)
Comment (cjsim89) [reply app/views/movies/show.html.erb:14]: You could also use a path like `"/actor_movies"`, and find a way to append the movie_id to the sent data... I can think of two ways to do that. 🤔

File: app/views/studios/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+<% @studios.each do |studio| %>
+  <div id="studio_<%= studio.id %>_info">
+    <h2><%= studio.name %></h2>
+    <p>Location: <%= studio.location %></p>
+
+    <div id="studio_<%= studio.id %>_movies" %>
+      <h3>Movies</h3>
+      <% studio.movies.each do |movie| %>
+        <div id="studio_<%= studio.id %>_movie_<%= movie.id %>" %>
```
Comment (cjsim89) [inline app/views/studios/index.html.erb:11]: Really nice use of IDs here to ensure you can find the relevant data.

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,18 @@
...
+@movie_1 = @studio_1.movies.create!(title: 'Movie Title 1', creation_year: '2000', genre: 'Movie Genre 1')
+@movie_2 = @studio_1.movies.create!(title: 'Movie Title 2', creation_year: '2005', genre: 'Movie Genre 1')
+@movie_3 = @studio_1.movies.create!(title: 'Movie Title 3', creation_year: '2015', genre: 'Movie Genre 2')
+@movie_4 = @studio_2.movies.create!(title: 'Movie Title 4', creation_year: '1956', genre: 'Movie Genre 3')
+@movie_5 = @studio_2.movies.create!(title: 'Movie Title 5', creation_year: '1974', genre: 'Movie Genre 1')
+@actor_1 = @movie_1.actors.create!(name: 'Actor Name 1', age: 43)
+@actor_2 = @movie_1.actors.create!(name: 'Actor Name 2', age: 25)
+@actor_3 = @movie_1.actors.create!(name: 'Actor Name 3', age: 30)
+@actor_4 = @movie_1.actors.create!(name: 'Actor Name 4', age: 76)
+@actor_5 = @movie_1.actors.create!(name: 'Actor Name 5', age: 50)
```
Comment (cjsim89) [inline db/seeds.rb:22]: Have you used the `faker` gem before? [Check it out](https://github.com/faker-ruby/faker), I think you'll enjoy it

File: spec/features/movies/show_spec.rb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+      expect(page).to have_content('Movie Title 1')
+      expect(page).to have_content('Creation Year: 2000')
+      expect(page).to have_content('Genre: Movie Genre 1')
+
+      expect(page).to_not have_content('Movie Title 2')
+
+      within '#movie_actors' do
+        expect('Actor Name 2').to appear_before('Actor Name 3')
+        expect('Actor Name 3').to appear_before('Actor Name 1')
```
Comment (cjsim89) [inline spec/features/movies/show_spec.rb:32]: Good use of within blocks as well as `orderly`.

**Review by cjsim89 — COMMENTED** [review]
Good job overall!

File: app/controllers/actor_movies_controller.rb (hunk 1 above)
Comment (JaredMHobson) [reply app/controllers/actor_movies_controller.rb:3]: That's interesting! I _think_ Abdul recommended the opposite and recommends only using the bang operator in your tests, not your actions/methods.

**Pull request closed without merging**